src/Testcontainers/Builders/ImageFromDockerfileBuilder.cs

```diff
@@ -75,6 +75,9 @@ public IImageFromDockerfileBuilder WithBuildArgument(string name, string value)
     /// <inheritdoc />
     public Task<string> Build()
     {
+      _ = Guard.Argument(this.DockerResourceConfiguration.DockerEndpointAuthConfig, nameof(IDockerResourceConfiguration.DockerEndpointAuthConfig))
```

Comment:
One small question: why discard assignment here and not just calling the function alone? that is why `_ =` at the front?

Reply:
If I undersand correctly it is new best practice to show that the result is useless for reader or even for compiler. Zou can find out more here: https://docs.microsoft.com/en-us/dotnet/csharp/fundamentals/functional/discards 
I would like to know if there is some real compiler optimiyation or is not.

Reply:
As far as I know IL is the same (no compiler optimization).